storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java

```diff
@@ -316,7 +317,7 @@ public void launchDaemon() {
             //This will only get updated once
             StormMetricsRegistry.registerMeter("supervisor:num-launched").mark();
             StormMetricsRegistry.registerMeter("supervisor:num-shell-exceptions", ShellUtils.numShellExceptions);
-            StormMetricsRegistry.startMetricsReporters(conf);
+            metricsReporters = StormMetricsRegistry.startMetricsReporters(conf);
```

Comment:
since you are not really returning metricsReporters here, better not to use this name `metricsReporters`